core/common/src/main/java/alluxio/collections/LockPool.java

```diff
@@ -245,37 +106,13 @@ public Optional<RWLockResource> tryGet(K key, LockMode mode) {
    */
   @VisibleForTesting
   public ReentrantReadWriteLock getRawReadWriteLock(K key) {
-    return mPool.getOrDefault(key, new Resource(new ReentrantReadWriteLock())).mLock;
-  }
-
-  private Resource getResource(K key) {
-    Preconditions.checkNotNull(key, "key can not be null");
-    Resource resource = mPool.compute(key, (k, v) -> {
-      if (v != null && v.mRefCount.incrementAndGet() > 0) {
-        // If the entry is to be removed, ref count will be INT_MIN, so incrementAndGet will < 0.
-        v.mIsAccessed = true;
-        return v;
-      }
-      return new Resource(mDefaultLoader.apply(k));
-    });
-    if (mPool.size() > mHighWatermark) {
-      if (mEvictLock.tryLock()) {
-        try {
-          mOverHighWatermark.signal();
-        } finally {
-          mEvictLock.unlock();
-        }
-      }
-    }
-    return resource;
+    return mCache.computeIfAbsent(key, (K) -> new ReentrantReadWriteLock());
```

Comment:
is it guaranteed we won't  get a null value if the underlying value is wrapped with weak ref?